aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -75,17 +75,17 @@ func resourceAwsKinesisFirehoseDeliveryStream() *schema.Resource {
 				},
 				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 					value := v.(string)
-					if value != "s3" && value != "redshift" && value != "elasticsearch" {
+					if value != "s3" && value != "extended_s3" && value != "redshift" && value != "elasticsearch" {
 						errors = append(errors, fmt.Errorf(
-							"%q must be one of 's3', 'redshift', 'elasticsearch'", k))
+							"%q must be one of 's3', 'extended_s3', 'redshift', 'elasticsearch'", k))
 					}
 					return
 				},
 			},
 
 			"s3_configuration": {
 				Type:     schema.TypeList,
-				Required: true,
+				Optional: true,
```

Comment:
I initially planned to mark s3_configuration as deprecated but decided against it because this structure is still used when destination is elasticsearch and redshift (see resourceAwsKinesisFirehoseDeliveryStreamCreate approx line 811)